src/views/Products.tsx

```diff
@@ -0,0 +1,123 @@
+import React, { useState, useEffect, useCallback } from 'react'
+import ProductsSidebar from '../components/ProductsSidebar'
+import ProductsJson from '../mocks/en-us/products.json'
+import '../styles/Products.scss'
+import ProductItem from '../components/FeaturedProducts/ProductItem'
+import ProductsPagination from '../components/ProductsPagination'
+import Spinner from '../components/Spinner'
+import { Link, useNavigate, useSearchParams } from 'react-router-dom'
+import { useProducts } from '../utils/hooks/useProducts'
+import { useSelector } from 'react-redux'
+
+export default function Products () {
+  const navigate = useNavigate()
+  const [searchParams] = useSearchParams()
+  const [filters, setFilters] = useState<Array<string>>([])
+  const { products, getProducts } = useProducts()
+  const {
+    data: categories,
+    isLoading
+  } = useSelector((state: any) => state.categories)
+
+  const handleCheckInput = (id: string|null, value: boolean = true) => {
+    const categoryToBeHandled: any = document.getElementById(`category-${id}`)
+    if (categoryToBeHandled) {
+      categoryToBeHandled.checked = value
+    }
+  }
+
+  const handleClearFilter = () => {
+    filters.forEach((filter) => handleCheckInput(filter, false))
+    setFilters([])
+    navigate('/products')
+    navigate(0)
+  }
+
+  const handleFilters = (e: Event) => {
+    const { checked, value } = (e.target as HTMLInputElement)
+    let categoriesValue: string = ''
+    if (checked) {
+      categoriesValue = searchParams.get('category')
+        ? `${searchParams.get('category')}|${value}`
+        : value
+      navigate(`/products?category=${categoriesValue}`)
+      navigate(0)
+    } else {
+      if (searchParams.get('category') && (searchParams.get('category')?.split('|')as any).length >= 1) {
+        const categoriesFilters = searchParams.get('category')?.split('|').filter((category) => category !== value)
+        categoriesFilters?.forEach((category, index) => {
+          if (categoriesFilters.length === index + 1) {
+            categoriesValue += `${category}`
+          } else {
+            categoriesValue += `${category}|`
+          }
+        })
+        navigate(`/products?category=${categoriesValue}`)
+        navigate(0)
+      }
+    }
+  }
+
+  const handleProductFetch = (page: number) => {
+    getProducts(page)
+  }
+
+  useEffect(() => {
+    if (searchParams.get('category')) {
+      const categoryId: string|null = searchParams.get('category')
+      if (categoryId?.includes('|')) {
+        categoryId.split('|').forEach((category) => {
+          handleCheckInput(category)
+          setFilters([...filters, category as string])
+        })
+      } else {
+        handleCheckInput(categoryId)
+        setFilters([...filters, categoryId as string])
+      }
+    }
```

Comment:
an alternative could look like:

```suggestion
    // get the cateegory from the params
    const categoryId = searchParams.get('category')
    if (!categoryId) {
      return
    }
    // inside this block category will always be a string
    if (categoryId?.includes('|')) {
      const categories = categoryId.split('|')
      for (const category of categories) {
        // handleCheckInput should not modify the DOM directly
        handleCheckInput(category)
      }
      setFilters([...filters, ...categories])
    } else {
      handleCheckInput(categoryId)
      setFilters([...filters, categoryId as string])
    }
```